### PR TITLE
Apply Bincrafters conventions

### PR DESCRIPTION
**Pull request by uilianries**
/cc @danimtb

ONLY PR CI JOBS ARE RUNNING

### CODE REVIEW

File: conanfile.py [hunk 1]
```diff
@@ -1,69 +1,78 @@
...
 
-        os.rename(extracted_dir, 'sources')
+    def _build_cmake(self):
+        cmake = self._configure_cmake()
+        cmake.build()
 
     def build(self):
+        tools.patch(base_path=self._source_subfolder, patch_file="czmq.diff")
         if self.settings.compiler == 'Visual Studio':
             with tools.vcvars(self.settings, force=True, filter_known_paths=False):
```
Comment (danimtb) [inline conanfile.py]: Is this really needed? Isn't vcvars called inside cmake?

**Review by danimtb — APPROVED** [review]
approved the changes. So much better. just check my comment, but probably that would be better to test/try it in another PR

File: conanfile.py (hunk 1 above)
Comment (SSE4) [reply conanfile.py]: previously, it wasn't called for Ninja/NMake generators
now it's fixed, and calling vcvars is no longer needed

Comment (danimtb) [reply conanfile.py]: Thanks for the clarification @SSE4

Comment (uilianries) [reply conanfile.py]: the patch is used to:
- remove "-O3" from cxxflags
- skip the tests. There is no option to build or not the tests.

**Review by uilianries — COMMENTED** [review]
LGTM

**Comment by SSE4** [comment]
it suddenly has started to fail on OSX:
```
/System/Library/Frameworks/Kernel.framework/Headers/mach/port.h:145:22: error: typedef redefinition with different types ('ipc_port_t' (aka 'struct ipc_port *') vs '__darwin_mach_port_t' (aka 'unsigned int'))
typedef ipc_port_t              mach_port_t;
                                ^
/System/Library/Frameworks/Kernel.framework/Headers/sys/_types/_mach_port_t.h:49:30: note: previous definition is here
typedef __darwin_mach_port_t mach_port_t;
```

**Comment by uilianries** [comment]
@SSE4 All green again!

**Pull request merged**